src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -253,9 +265,10 @@ public virtual ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int a
 
             NamedTypeSymbol? namedType = null;
 
-            ImmutableArray<NamedTypeSymbol> namespaceOrTypeMembers;
+            Debug.Assert(scope is NamespaceSymbol or NamedTypeSymbol);
```

Comment:
>Debug.Assert(scope is NamespaceSymbol or NamedTypeSymbol); [](http://example.com/codeflow?start=12&length=58)

Consider putting this assert at the top, next to ```Debug.Assert(scope is not MergedNamespaceSymbol);```

Reply:
done.